Dockerfile

```diff
@@ -2,6 +2,11 @@ FROM --platform=$BUILDPLATFORM python:3.11-slim-bookworm
 ARG TARGETPLATFORM
 ARG BUILDPLATFORM
 ARG TARGETARCH
+ARG ARTIFACTORY_USR
+ARG ARTIFACTORY_PSW
+ENV POETRY_HTTP_BASIC_DS1_PYPI_USERNAME="${ARTIFACTORY_USR}" \
+    POETRY_HTTP_BASIC_DS1_PYPI_PASSWORD="${ARTIFACTORY_PSW}"
```

Comment:
```suggestion
ARG ARTIFACTORY_USR
ARG ARTIFACTORY_PSW
ENV POETRY_HTTP_BASIC_DS1_PYPI_USERNAME="${ARTIFACTORY_USR}" \
    POETRY_HTTP_BASIC_DS1_PYPI_PASSWORD="${ARTIFACTORY_PSW}"
```

Could we make this agnostic to hosting of artifacts (semantics)? I don't see any references to these vars either, so this maybe is just a runtime thing? Here is an example of how I handled runtime for node. https://github.com/DatasiteLabs/docker-cicd-node/blob/main/inject-npmrc.sh

I think we can solve at runtime instead of at the container level. 

<img width="971" alt="image" src="https://github.com/DatasiteLabs/docker-cicd-python/assets/5103752/3a98f081-b295-403f-80d2-901aad1f01f5">